pkg/pool-weighted/contracts/LBPool.sol

```diff
@@ -0,0 +1,294 @@
+
+    PoolState private _poolState;
+
+    /**
+     * @notice Emitted when the owner enables or disables swaps.
+     * @param swapEnabled True if we are enabling swaps
+     */
+    event SwapEnabledSet(bool swapEnabled);
+
+    /**
+     * @notice Emitted when the owner initiates a gradual weight change (e.g., at the start of the sale).
+     * @dev Also emitted on deployment, recording the initial state.
+     * @param startTime The starting timestamp of the update
+     * @param endTime  The ending timestamp of the update
+     * @param startWeights The weights at the start of the update
+     * @param endWeights The final weights after the update is completed
+     */
+    event GradualWeightUpdateScheduled(
+        uint256 startTime,
+        uint256 endTime,
+        uint256[] startWeights,
+        uint256[] endWeights
+    );
+
+    /// @dev Indicates that the router that called the Vault is not trusted, so any operations should revert.
+    error RouterNotTrusted();
+
+    constructor(
+        NewPoolParams memory params,
+        IVault vault,
+        address owner,
+        bool swapEnabledOnStart,
+        address trustedRouter
+    ) WeightedPool(params, vault) Ownable(owner) {
+        // WeightedPool validates `numTokens == normalizedWeights.length`, and ensures valid weights.
+        // Here we additionally enforce that LBPs must be two-token pools.
+        InputHelpers.ensureInputLengthMatch(_NUM_TOKENS, params.numTokens);
+
+        // Set the trusted router (passed down from the factory).
+        _TRUSTED_ROUTER = trustedRouter;
+
+        // solhint-disable-next-line not-rely-on-time
+        uint32 currentTime = block.timestamp.toUint32();
+        _startGradualWeightChange(currentTime, currentTime, params.normalizedWeights, params.normalizedWeights);
+        _setSwapEnabled(swapEnabledOnStart);
+    }
+
+    /**
+     * @notice Return start time, end time, and endWeights as an array.
+     * @dev Current weights should be retrieved via `getNormalizedWeights()`.
+     * @return startTime The starting timestamp of any ongoing weight change
+     * @return endTime The ending timestamp of any ongoing weight change
+     * @return endWeights The "destination" weights, sorted in token registration order
+     */
+    function getGradualWeightUpdateParams()
+        external
+        view
+        returns (uint256 startTime, uint256 endTime, uint256[] memory endWeights)
+    {
+        PoolState memory poolState = _poolState;
+
+        startTime = poolState.startTime;
+        endTime = poolState.endTime;
+
+        endWeights = new uint256[](_NUM_TOKENS);
+        endWeights[0] = poolState.endWeight0;
+        endWeights[1] = FixedPoint.ONE - poolState.endWeight0;
+    }
+
+    /**
+     * @notice Indicate whether swaps are enabled or not for the given pool.
+     * @return swapEnabled True if trading is enabled
+     */
+    function getSwapEnabled() external view returns (bool) {
```

Comment:
It's probably not worth it, but this is a type of "pause" and we could consider adding a "swapsPaused" state to the vault. The goal of the pause manager was to have a consistent pattern for all pools when pausing, so we have less instances where internally the pool disables itself and we have to react to that state change. It increases operational burden for all of the off-chain tooling.

Reply:
I'd hesitate to add something to the Vault to accommodate a specialized kind of pool (and all the checks through the Vault code; unbalanced add is a swap, etc.)

We could just use the regular pause, if we don't think the owner needs to add liquidity while swaps are paused. Of course, it would complicate the initialization, as they wouldn't be able to fund the pool if it starts paused. Or they'd have to unpause/initialize/pause in one transaction.

Reply:
I would lean towards keeping it as it. Using the existing notion of paused pools feels like trying to shoe-horn in something that's similar but not exactly what we need. The unpause/initialize/pause seems like bending over backwards just to use that tool. I do also agree with @EndymionJkb that adding different `<Something>Managers` could spiral out of control. I think it's evident enough how it should work w/ the pool `owner`, but it someone strongly disagrees, I'm happy to discuss.